src/ComparedQueryable/QueryableExtensions.cs

```diff
@@ -11,25 +12,32 @@ public static class QueryableExtensions
         /// properites of which can be naturally sorted.
         /// </summary>
         /// <typeparam name="T"></typeparam>
-        /// <param name="collection"></param>
+        /// <param name="source"></param>
         /// <returns></returns>
-        public static IQueryable<T> AsNaturalQueryable<T>(this IEnumerable<T> collection)
+        public static IQueryable<T> AsNaturalQueryable<T>(this IEnumerable<T> source)
         {
-            return collection.AsComparedQueryable(NaturalSortComparer.Instance);
+            return source.AsComparedQueryable(NaturalSortComparer.Instance);
         }
 
         /// <summary>
         /// Converts an <see cref="IEnumerable{T}"/> into an <see cref="IQueryable{T}"/>.
         /// </summary>
         /// <typeparam name="T"></typeparam>
         /// <typeparam name="TComparison">Will order all properties of this type within <see cref="T"/>.</typeparam>
-        /// <param name="collection"></param>
+        /// <param name="source"></param>
         /// <param name="comparer">Used for ordering operations.</param>
         /// <returns></returns>
-        public static IQueryable<T> AsComparedQueryable<T, TComparison>(this IEnumerable<T> collection,
+        public static IQueryable<T> AsComparedQueryable<T, TComparison>(this IEnumerable<T> source,
             IComparer<TComparison> comparer)
         {
-            return new EnumerableQuery<T, TComparison>(collection, comparer);
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));

```

Comment:
We could. It's so weird to expect normal behavior from a null collection, I'm not sure if there's much value in it.